swap/src/cli/watcher.rs

```diff
@@ -45,6 +48,25 @@ impl Watcher {
         loop {
             tokio::time::sleep(Duration::from_secs(Watcher::CHECK_INTERVAL)).await;
 
+            // Fetch the current Bitcoin balance
+            match self.wallet.balance().await {
+                Ok(new_balance) => {
+                    // Check if the balance has changed
+                    if new_balance != self.balance {
```

Comment:
I'd also be fine with sending this regardless of if the balance changed. It'd simplify this a little bit and provide some redundancy (e.g if the fronted missed the first event due to not having fully loaded yet)

The events are sent over IPC so performance shouldn't be a big issue.

Reply:
makes sense, i'll do that